app/collectors/reservation.scala

```diff
@@ -0,0 +1,56 @@
+package collectors
+
+import agent._
+import com.amazonaws.services.ec2.AmazonEC2Client
+import com.amazonaws.services.ec2.model.{DescribeReservedInstancesRequest, ReservedInstances}
+import controllers.routes
+import org.joda.time.{DateTime, Duration}
+import play.api.mvc.Call
+import utils.Logging
+
+import collection.JavaConverters._
+import scala.util.Try
+
+object ReservationCollectorSet extends CollectorSet[Reservation](ResourceType("reservations", Duration.standardMinutes(15L))) {
+  val lookupCollector: PartialFunction[Origin, Collector[Reservation]] = {
+    case amazon: AmazonOrigin => AWSReservationCollector(amazon, resource)
+  }
+}
+
+case class AWSReservationCollector(origin: AmazonOrigin, resource: ResourceType) extends Collector[Reservation] with Logging {
+
+  val client = new AmazonEC2Client(origin.credentials.provider)
+  client.setRegion(origin.awsRegion)
+
+  def crawl: Iterable[Reservation] = {
+    client.describeReservedInstances(new DescribeReservedInstancesRequest()).getReservedInstances.asScala.map {
+      Reservation.fromApiData
+    }
+  }
+}
+
+case class Reservation(
+  key: String,
+  region: String,
+  instanceType: String,
+  instanceCount: Int,
+  startTime: Option[DateTime],
+  endTime: Option[DateTime]
+) extends IndexedItem {
+  override def arn: String = s"arn:gu:reservation:key/$key"
```

Comment:
Whilst there isn't a real ARN format for reservations I suggest we use something closer to what we imagine it to be. Perhaps:
`arn:aws:ec2:<region>:<accountId>:reservation/<id>`